sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,75 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
 {
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  std::string name;
+  std::string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(std::string name, std::string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  std::string getName();
+  std::string getValue();
 };
 
-http_request::http_request() {}
-http_request::~http_request() {}
+class Request
+{
+
+private:
+  http_method::HttpMethod method;
+  std::string url;
+  std::vector<Header> headers;
+  std::string body;
+  size_t query_start; // 0 = no query in url, > 0 = query start position in url '?'
+
+  inline size_t get_query_start(std::string someUrl)
+  {
+    auto position = someUrl.find('?');
+    return position == std::string::npos ? 0 : position;
+  }
+
+public:
+  Request(http_method::HttpMethod httpMethod, std::string url)
+  {
+    this->method = httpMethod;
+    this->url = url;
+    this->query_start = get_query_start(url);
+  }
+
+  Request(http_method::HttpMethod httpMethod, std::string url, std::string body) : Request(httpMethod, url)
+  {
+    this->body = body;
```

Comment:
I didn't read through all the code. But is this `body` the whole HTTP request body?

There's a new feature Jumbo Block in STG73, where request body can be up to 4GB and probably 40GB in the future. In this case, I don't think we can assume the whole body can be loaded into memory.

Reply:
OK, let's review how we want to handle body in next meeting.
I believe we can have the basic case with just an std::string and set a MAX length for it.
Then we can have the advance case for request greater than that.
But, I would wait for what @JeffreyRichter has done for other languages in the past.